eclass/java-utils-2.eclass

```diff
@@ -2945,7 +2943,14 @@ is-java-strict() {
 java-pkg_clean() {
 	NO_DELETE=()
 	for keep in ${JAVA_PKG_NO_CLEAN[@]}; do
-		NO_DELETE+=( '!' '-wholename' ${keep} )
+		if [[ "${keep%.jar}" != ${keep} ]]; then
+			NO_DELETE+=( '!' '-name' ${keep} )
```

Comment:
i'm not sure this covers all cases as `find` documentation says about `-name` this: `Base  of  file  name  (the  path with the leading directories removed) matches shell pattern pattern....`. that would mean that `name` should be just a file name, not a path, but this script does not attempt to find out whether it's just a file name or a path.

Reply:
> [...] should be just a file name, not a path, [...]

Just added `'-type f'` to those lines.

Reply:
do we need `-name` specified in all those cases? i think `-path` pattern would be enough for handling all the cases. it would require specifying the whole pattern, but that can be also taken as a safety precaution.

Reply:
i think specifying just file name is a bit risky, as the same file name might be in several places and only one might be the target for preserving. so probably the previous (already committed) patch is enough wrt functionality.

Reply:
> might be in several places and only one might be the target for preserving

This finally convinces me. So I dropped that commit.